packages/cloud/src/libraries/services.ts

```diff
@@ -13,4 +19,80 @@ export class ServicesLibrary {
 
     return application.customClientMetadata.tenantId;
   }
+
+  async getAdminTenantLogtoConnectors(): Promise<LogtoConnector[]> {
+    const databaseConnectors = await this.queries.connectors.findAllConnectors(adminTenantId);
+
+    const logtoConnectors = await Promise.all(
+      databaseConnectors.map(async (databaseConnector) => {
+        const { id, metadata, connectorId } = databaseConnector;
+
+        const connectorFactories = await loadConnectorFactories();
+
+        const connectorFactory = connectorFactories.find(
+          ({ metadata }) => metadata.id === connectorId
+        );
+
+        if (!connectorFactory) {
+          return;
+        }
+
+        try {
+          const { rawConnector, rawMetadata } = await buildRawConnector(
+            connectorFactory,
+            async () => {
+              const databaseConnectors = await this.queries.connectors.findAllConnectors(
+                adminTenantId
+              );
+              const connector = databaseConnectors.find((connector) => connector.id === id);
+
+              if (!connector) {
+                throw new RequestError(`Unable to find connector ${id}`, 500);
+              }
+
+              return connector.config;
+            }
+          );
```

Comment:
you can use `trySafe` to wrap this section to make it cleaner

Reply:
updated